src/ssh/cockpitsshrelay.c

```diff
@@ -896,19 +896,18 @@ do_password_auth (CockpitSshData *data)
   return rc;
 }
 
+struct CockpitSshPromptData {
```

Comment:
It's `ssh_userauth_publickey_auto`: http://api.libssh.org/stable/group__libssh__auth.html#ga53e6771b250c061463ed98b6e5b6e0af

I didn't try to use it as part of this PR since it felt like a big distraction at the point, but now is a different time and we might just give it a shot.

Reply:
See #14541 for a experiment of trying to use that function. Result: we shouldn't yet since the workarounds needed to make it work are worse than having the hardcoded list here. I'll update the comments in the code to explain the status of this hack.

Reply:
#14551 has taken care of this.